pkg/k8s/watchers/pod.go

```diff
@@ -721,6 +727,10 @@ func (k *K8sPodWatcher) genServiceMappings(pod *slim_corev1.Pod, podIPs []string
 				nodeAddrAll = append(nodeAddrAll, netip.IPv4Unspecified())
 				nodeAddrAll = append(nodeAddrAll, netip.IPv6Unspecified())
 			}
+			if !loopbackHostport && ip.ListContainsAddr(nodeAddrAll, ip.MustAddrFromIP(hostIP)) {
```

Comment:
Is there a reason you are checking HostIP and not NodeName?

Reply:
It is more reasonable to use node name.